go/parquet/pqarrow/encode_arrow.go

```diff
@@ -607,6 +607,30 @@ func writeDenseArrow(ctx *arrowWriteContext, cw file.ColumnChunkWriter, leafArr
 				}
 				wr.WriteBatchSpaced(data, defLevels, repLevels, arr.NullBitmapBytes(), int64(arr.Data().Offset()))
 			}
+		case *arrow.MonthDayNanoIntervalType:
+			arr := leafArr.(*array.MonthDayNanoInterval)
+			data := make([]parquet.FixedLenByteArray, arr.Len())
+
+			// NOTE: parquet Interval only supports 32-bit milliseconds so we are truncating Nanoseconds to milliseconds
+			if arr.NullN() == 0 {
+				for idx := range data {
+					data[idx] = make(parquet.FixedLenByteArray, 12)
+					binary.LittleEndian.PutUint32(data[idx][:4], uint32(arr.Value(idx).Months))
```

Comment:
I think there may be more logic involved in this conversion. The arrow values for each of these are [signed](https://github.com/apache/arrow/blob/main/format/Schema.fbs#L409-L415) integers, which means the conversion would need to account for negative values as well.

Reply:
You are right. I somehow kept assuming that intervals cannot be negative.  
In this case, I think it is not possible to convert arrow MonthDayNanoInterval into parquet Interval.  
I'll close this then.